File: dotnet/DesignPatterns/BehavioralDesignPatterns/ObserverDesignPattern/ObserverDesignPattern.Example2/CurrentConditionsDisplay.cs
namespace ObserverDesignPattern.Example2;

public class CurrentConditionsDisplay : IObserver
{
    private float temperature;
    private float humidity;
    private ISubject _weatherStation;

    public CurrentConditionsDisplay(ISubject weatherStation)
    {
        _weatherStation = weatherStation;
        _weatherStation.RegisterObserver(this);
    }

    public void Update(float temperature, float humidity, float pressure)
    {
        this.temperature = temperature;
        this.humidity = humidity;
        Display();
    }

    public void Display()
    {
        Console.WriteLine($"🌡️  Current Conditions: {temperature}°C, Humidity: {humidity}%");
    }
}

File: nodejs/DesignPatterns/BehavioralDesignPatterns/ObserverDesignPattern/ObserverDesignPattern.Example2/src/CurrentConditionsDisplay.ts
import { IObserver } from './IObserver';
import { ISubject } from './ISubject';

export class CurrentConditionsDisplay implements IObserver {
  private _temperature: number = 0;
  private _humidity: number = 0;
  private readonly _weatherStation: ISubject;

  constructor(weatherStation: ISubject) {
    this._weatherStation = weatherStation;
    this._weatherStation.registerObserver(this);
  }

  update(temperature: number, humidity: number, _pressure: number): void {
    this._temperature = temperature;
    this._humidity = humidity;
    this.display();
  }

  display(): void {
    console.log(`🌡️  Current Conditions: ${this._temperature}°C, Humidity: ${this._humidity}%`);
  }
}
